plugins/app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/location/MyLocationModesActivity.java

```diff
@@ -0,0 +1,132 @@
+package com.mapbox.mapboxsdk.plugins.testapp.activity.location;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.plugins.mylocationlayer.MyLocationLayerMode;
+import com.mapbox.mapboxsdk.plugins.mylocationlayer.MyLocationLayerOptions;
+import com.mapbox.mapboxsdk.plugins.mylocationlayer.MyLocationLayerPlugin;
+import com.mapbox.mapboxsdk.plugins.testapp.R;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+public class MyLocationModesActivity extends AppCompatActivity implements OnMapReadyCallback {
+
+  @BindView(R.id.mapView)
+  MapView mapView;
+  @BindView(R.id.button_location_mode_none)
+  Button locationModeNoneButton;
+  @BindView(R.id.button_location_mode_compass)
+  Button locationModeCompassButton;
+  @BindView(R.id.button_location_mode_tracking)
+  Button locationModeTrackingButton;
+  @BindView(R.id.button_location_mode_navigation)
+  Button locationModeNavigationButton;
+
+  private MyLocationLayerOptions options;
```

Comment:
Do we really want an options object to configure a plugin? The options paradigm, that is used in the main SDK was introduced to separate setters/getters from MapboxMap and inspired by similar constructs in the Gmaps SDK. Since the Location plugin is standalone. I don't feel we need a separate options object to make it more complicated.